packages/circuit-ui/components/Button/Button.tsx

```diff
@@ -391,17 +403,20 @@ export const Button = forwardRef(
         as={props.href ? Link : 'button'}
         onClick={handleClick}
       >
-        {Icon && <Icon css={iconStyles} role="presentation" />}
-        {loadingLabel ? (
-          <Fragment>
-            <LoadingIcon isLoading={Boolean(isLoading)} size="byte">
-              <LoadingLabel>{loadingLabel}</LoadingLabel>
-            </LoadingIcon>
-            <Children isLoading={Boolean(isLoading)}>{children}</Children>
-          </Fragment>
-        ) : (
-          children
-        )}
+        <Fragment>
+          {Icon && (
+            <StyledIcon
+              // @ts-expect-error This is gonna be addressed in v4.
+              as={Icon}
+              isLoading={Boolean(isLoading)}
+              role="presentation"
+            />
+          )}
+          <LoadingIcon isLoading={Boolean(isLoading)} size="byte">
+            <LoadingLabel>{loadingLabel}</LoadingLabel>
+          </LoadingIcon>
+          <Content isLoading={Boolean(isLoading)}>{children}</Content>
+        </Fragment>
```

Comment:
The Icon should be wrapped in the Content component together with the `children`. This would remove the need for loading styles on the Icon.

Reply:
Updated the implementation here [c636685](https://github.com/sumup-oss/circuit-ui/pull/1167/commits/c63668568cbe0247687a82e840aab5c407fd4c7b)